lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -599,6 +599,31 @@ struct PathOpConversion : public OpConversionPattern<firrtl::PathOp> {
   const PathInfoTable &pathInfoTable;
 };
 
+struct WireOpConversion : public OpConversionPattern<WireOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(WireOp wireOp, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto wireValue = dyn_cast<FIRRTLPropertyValue>(wireOp.getResult());
+
+    // If the wire isn't a Property, not much we can do here.
+    if (!wireValue)
+      return failure();
+
+    // Find the assignment to the wire.
+    PropAssignOp propAssign = getPropertyAssignment(wireValue);
```

Comment:
This is safe (no concerns about dominance or def-before-use ordering) because this is only applying to code in a graph region, is that correct?

Reply:
Ok this is definitely important that we are in a graph region at this point... if we were in a SSACFG region we couldn't do this so trivially. I'm adding a check to this pattern to confirm we're in a graph region before doing the trivial rewrite.